superset/assets/src/dashboardList/DashboardList.tsx

```diff
@@ -0,0 +1,243 @@
+          >
+            <i className="fa fa-trash" />
+          </span>
+          <span
+            role="button"
+            className="action-button"
+            onClick={() => this.handleDashboardEdit(original)}
+          >
+            <i className="fa fa-pencil" />
+          </span>
+        </span>
+      ),
+    },
+  ];
+
+  initialSort = [{ id: 'changed_on', desc: true }];
+
+  handleDashboardEdit = ({ id }: { id: number }) => {
+    window.location.assign(`/dashboard/edit/${id}`);
+  };
+
+  handleDashboardDeleteConfirm = (dashboard: any) => {
+    this.setState({
+      deleteCandidate: dashboard,
+      showDeleteModal: true,
+    });
+  };
+
+  handleDashboardDelete = () => {
+    const { id, title } = this.state.deleteCandidate;
+    SupersetClient.delete({
+      endpoint: `/api/v1/dashboard/${id}`,
+    }).then(
+      resp => {
+        console.log(resp);
+        const dashboards = this.state.dashboards.filter(d => d.id !== id);
+        this.setState({
+          dashboards,
+          showDeleteModal: false,
+          deleteCandidate: {},
+        });
+      },
+      (err: any) => {
+        this.props.addDangerToast(t(`There was an issue deleting ${title}`));
+        this.setState({ showDeleteModal: false, deleteCandidate: {} });
+      },
+    );
+  };
+
+  toggleModal = () => {
+    this.setState({ showDeleteModal: !this.state.showDeleteModal });
+  };
+
+  fetchData = ({ pageIndex, pageSize, sortBy, filters }) => {
+    this.setState({ loading: true });
+    const filterExps = Object.keys(filters).map(fk => ({
+      col: fk,
+      opr: filters[fk].filterId,
+      value: filters[fk].filterValue,
+    }));
+
+    const queryParams = JSON.stringify({
+      order_column: sortBy[0].id,
+      order_direction: sortBy[0].desc ? 'desc' : 'asc',
+      page: pageIndex,
+      page_size: pageSize,
+      ...(filterExps.length ? { filters: filterExps } : {}),
+    });
+
+    return SupersetClient.get({
+      endpoint: `/api/v1/dashboard/?q=${queryParams}`,
+    })
+      .then(({ json }) => {
+        this.setState({ dashboards: json.result, dashboard_count: json.count });
+      })
+      .catch(() => {
+        this.props.addDangerToast(
+          t('An error occurred while fetching Dashboards'),
+        );
+      })
+      .finally(() => this.setState({ loading: false }));
+  };
+
+  render() {
+    return (
+      <div className="container welcome">
+        <Panel>
+          <ListView
+            className="dashboard-list-view"
+            title={'Dashboards'}
+            columns={this.columns}
+            data={this.state.dashboards}
+            count={this.state.dashboard_count}
+            pageSize={PAGE_SIZE}
+            fetchData={this.fetchData}
+            loading={this.state.loading}
+            initialSort={this.initialSort}
+            filterable
+            filterTypes={[
+              { label: 'Starts With', value: 'sw' },
```

Comment:
This object can be fetched from the API and you get translations for free also. This is on the `_info` endpoint